src/DIRAC/WorkloadManagementSystem/Service/PilotManagerHandler.py

```diff
@@ -100,13 +101,79 @@ def export_addPilotTQReference(cls, pilotRef, taskQueueID, ownerDN, ownerGroup,
   ##############################################################################
   types_getPilotOutput = [six.string_types]
 
-  @staticmethod
+  @classmethod
   def export_getPilotOutput(pilotReference):
     """ Get the pilot job standard output and standard error files for the Grid
         job reference
     """
+    result = cls.pilotAgentsDB.getPilotInfo(pilotReference)
```

Comment:
```suggestion
    result = self.pilotAgentsDB.getPilotInfo(pilotReference)
```
And just a normal method, neither class nor static but with `self` as first argument.

Reply:
I am ok with this, but do you know why all the other methods of `PilotManagerHandler` are class methods? Is there any specific reason?

Reply:
I don't know why the other `export_`s are classmethods, I don't think they need to be (other services have `export_` as instance methods, e.g., ConfigurationHandler) and I think the export functions are called with the instance of the handler

https://github.com/DIRACGrid/DIRAC/blob/cbce4f036dda83159d5655e255ffddfc20a847de/src/DIRAC/Core/DISET/RequestHandler.py#L258-L262

So it seems to me classmethods aren't the thing to be used, but I could be missing some point about static/class/instance variables or inheritence or all of it

Reply:
You don't really *have to* use class methods.